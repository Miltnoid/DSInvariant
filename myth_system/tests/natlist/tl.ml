type nat =
  | O
  | S of nat

type natlist =
  | Nil
  | Cons of nat * natlist

let tl : natlist -> natlist |>
  { [] => []
  | [0] => []
  | [0; 0] => [0] } = ?
