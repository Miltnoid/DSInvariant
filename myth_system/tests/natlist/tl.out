(* Generated Wed Dec 31 11:16:17 2014 *)
let tl : natlist -> natlist =
  fun (n1:natlist) -> match n1 with
                        | Nil -> Nil
                        | Cons (n2, n3) -> n3
;;
